pkg/resource/service.go

```diff
@@ -0,0 +1,53 @@
+package resource
+
+import (
+	"context"
+	"github.com/odpf/entropy/domain"
+	"github.com/odpf/entropy/store"
+)
+
+type ServiceInterface interface {
+	CreateResource(ctx context.Context, res *domain.Resource) (*domain.Resource, error)
+	UpdateResource(ctx context.Context, urn string, configs map[string]interface{}) (*domain.Resource, error)
+}
+
+type Service struct {
+	resourceRepository store.ResourceRepository
+}
+
+func NewService(repository store.ResourceRepository) *Service {
+	return &Service{
+		resourceRepository: repository,
+	}
+}
+
+func (s *Service) CreateResource(ctx context.Context, res *domain.Resource) (*domain.Resource, error) {
+	res.Urn = domain.GenerateResourceUrn(res)
+	res.Status = "PENDING"
```

Comment:
let's put it into separate const. WDYT?

Reply:
Yes it does. Changing the Protobuf definition as well for the same.